File: leetcode/py/1. Two Sum.py
# O(n) Time
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        #定义一个差值字典，key为差值，value为index
        jian_dict = {}
        for i in range(len(nums)):
            if nums[i] in jian_dict:
                return [jian_dict[nums[i]], i]
            else:
                jian_dict[target-nums[i]] = i

File: first_time/leetcode/c++/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> dict;
        vector<int> index;
        for(int i=0; i<nums.size(); ++i){
            if(dict.count(target - nums[i])){
                index.push_back(dict[target-nums[i]]);
                index.push_back(i);
                return index;
            }
            dict[nums[i]] = i;
        }
        return index;
    }
};